src/System.CommandLine/Parsing/CommandResult.cs

```diff
@@ -165,15 +210,16 @@ private void ValidateArguments(bool completeValidation)
                 {
                     argumentResult = (ArgumentResult)symbolResult;
                 }
-                else if (argument.HasDefaultValue)
+                else if (argument.HasDefaultValue || argument.Arity.MinimumNumberOfValues > 0)
                 {
                     argumentResult = new ArgumentResult(argument, SymbolResultTree, this);
                     SymbolResultTree[argument] = argumentResult;
-                }
-                else if (argument.Arity.MinimumNumberOfValues > 0)
-                {
-                    AddError(LocalizationResources.RequiredArgumentMissing(this));
-                    continue;
+
+                    if (!argument.HasDefaultValue && argument.Arity.MinimumNumberOfValues > 0)
+                    {
+                        argumentResult.AddError(LocalizationResources.RequiredArgumentMissing(argumentResult));
```

Comment:
this fixes the issue I've mentioned in https://github.com/dotnet/command-line-api/issues/2082#issuecomment-1460476519 : when a required argument is missing, we report the error for the ArgumentResult, not for CommandResult